plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/typehints/RequestCheckEdgeTargetActionHandler.java

```diff
@@ -0,0 +1,75 @@
+/********************************************************************************
+ * Copyright (c) 2023 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.server.features.typehints;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.glsp.graph.GModelElement;
+import org.eclipse.glsp.server.actions.AbstractActionHandler;
+import org.eclipse.glsp.server.actions.Action;
+import org.eclipse.glsp.server.diagram.DiagramConfiguration;
+import org.eclipse.glsp.server.model.GModelState;
+
+import com.google.inject.Inject;
+
+/**
+ * Default handler implementation for {@link RequestCheckEdgeAction}.
+ * Delegates the edge check to an {@link EdegeCreationChecker} if the given edge information is applicable
+ * (i.e. a checker is bound,the given edge type has a dynamic type hint and source/target elemnent are present in the
+ * diagram).
+ * Otherwise returns `false`.
+ */
+public class RequestCheckEdgeTargetActionHandler extends AbstractActionHandler<RequestCheckEdgeAction> {
+
+   @Inject
+   protected Optional<EdegeCreationChecker> edgeTargetChecker;
+
+   @Inject
+   protected DiagramConfiguration diagramConfiguration;
+
+   @Inject
+   protected GModelState modelState;
+
+   @Override
+   protected List<Action> executeAction(final RequestCheckEdgeAction action) {
+      return List.of(new CheckEdgeTargetResultAction(validate(action), action));
+   }
+
+   protected boolean validate(final RequestCheckEdgeAction action) {
+      if (!edgeTargetChecker.isPresent()) {
+         return false;
```

Comment:
Is this a reasonable default? I'm just wondering cause intuitively I'd say no because it is an optional service and not implementing an optional service should not stop something from working. However, I saw that in the client code we only ever send the action if we already know that the edge type hint is dynamic so we have a guard on that side. I'm not totally sure I think that should be sufficient, what do you think?